### ISSUE TITLE
HA Kubernetes Setup

### ISSUE THREAD

**Issue by capgadsx** [ChileanVirtualObservatory/jovial.chivo.cl#5]
The idea is to implement the solution described in the official kubernetes [documentation](https://kubernetes.io/docs/admin/high-availability/), using a clustered etcd deployment and load-balanced api-servers.
![Clusterd Kubernetes](https://d33wubrfki0l68.cloudfront.net/2555d34e3008aab4b049ca5634cfabc2078ccf92/3269a/images/docs/ha.svg)

**Comment by mosoriob** [comment]
Maybe this is a good start. https://github.com/kubernetes/kubeadm/issues/277

**Comment by madiazp** [comment]
We were thinking with @capgadsx  that it will be a good solution to put kubemaster and etcd in a VM in Ovirt and do backups in lustre with rsync, so the HA is up to Ovirt.  With a good VM configuration(auto net, auto service up in restarts, etc) the only enemy will be a IML server's failure or an entire datacenter failure.

**Comment by mosoriob** [comment]
That is overthink. 3 etcd is enough.

**Comment by maray** [comment]
I agree with @sirspock. Please do not complicate things more than it will be needed. If 3 independent disks of 3 different servers fail at the same time, everyone will understand a 503!

@capgadsx can you edit your first comment to add a brief doc of the 3 etcd solution we are currently using? (just for the sake of documentation)

**Comment by capgadsx** [comment]
At this point we already implemented the 3-member etcd cluster and we have 3 kubernetes api-servers running. We still have to setup the load-balancer for the api-servers
